### PR TITLE
webview improvements

### LINKED ISSUE #12: Generate overloads for event handlers
> Generate overloads for `once`, `addListener`, and `removeListener` to allow typesafe event handling. This is already being done for the `on` method.

@miniak suggested this. I believe @felixrieseberg said this is already covered...

### PR DESCRIPTION
**Pull request by zeke**
courtesy of @MarshallOfSound

Fixes #12

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
Will resolve conflicts on this today/tomorrow (what do you call today when it's 2am 🌵 )

**Comment by zeke** [comment]
All this action is on this PR now. :)

cc @miniak, @felixrieseberg, @omichelsen, @cifratila

**Comment by zeke** [comment]
That github-hosted `types-publisher` dep is very particular:

> error types-publisher@0.0.0: The engine "node" is incompatible with this module. Expected version ">=6.10.0".

Gonna have to drop that dep in order to get CI passing and be able to use this in electron/electron.

File: lib/module-declaration.js [hunk 1]
```diff
@@ -31,47 +32,68 @@ const generateModuleDeclaration = (module, index, API) => {
...
-        if (moduleEventListenerArg.description) {
-          argString += utils.wrapComment(moduleEventListenerArg.description).map((l, i) => `${l}
-${indent}`).join('')
+        if (eventListenerArg.description) {
+          argString += `\n${indent}`
+          argString += utils.wrapComment(eventListenerArg.description).map((l, i) => `${l}\n${indent}`).join('')
+        }
+
+        // This is a temporary fix
+        if (moduleEvent.name === 'login' && eventListenerArg.name === 'callback' && !eventListenerArg.parameters) {
```
Comment (MarshallOfSound) [inline lib/module-declaration.js]: 😢 I assume this just requires a change in the docs in Electron ❓

**Pull request merged**